src/components/Drawer/Drawer.component.tsx

```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+import { DisclosureProps, Disclosure, DisclosureContent, useDisclosureState } from 'reakit';
+
+import * as S from './Drawer.style';
+
+export type DrawerProps = DisclosureProps & {
+	disclosure: React.ComponentPropsWithRef<any>;
+	heading?: React.ComponentPropsWithRef<any>;
+	footer?: React.ComponentPropsWithRef<any>;
+	visible: boolean;
+};
+
+export const Drawer = React.forwardRef<React.ReactElement, React.PropsWithChildren<any>>(
+	({ disclosure, heading, children, footer, visible, ...props }: DrawerProps, ref) => {
+		const state = useDisclosureState({ animated: true, visible });
+
+		React.useEffect(() => {
+			state.setVisible(visible);
+		}, [visible]);
+
+		return (
+			<>
+				{disclosure && (
+					<Disclosure as={S.DrawerDisclosure} {...disclosure} {...disclosure.props}>
+						{disclosureProps => React.cloneElement(disclosure, disclosureProps)}
+					</Disclosure>
```

Comment:
This seems weird. 
- This `as` on `Disclosure` could potentially style it and change its HTML output
- However inside of it, as a children, you clone any component passed as a props

As a result you may get the `Disclosure` HTML output (a button as a default) and inside of that button you will have *anything* (possibly another button).

Don't we need to simply
```suggestion
					<Disclosure as={disclosure} {...disclosure} {...disclosure.props} />
```
instead?